### ISSUE TITLE
Miller Document API include / exclude elements without dates

### ISSUE THREAD

**Issue by gffuma** [C2DH/minett-stories#28]
How to  include / exclude elements without dates with Miller API?

**Comment by danieleguido** [comment]
hi @gffuma normally for instance own fields we use the notation `isnull` like: `/api/document/?filters={"url__isnull":true}`
For the posgres JSON field `data` you can chain children (operator is `AND`) . As an example, this request gives you back all documents that do not contains a FR translation of the date field, but an EN one:
```
/api/document/?filters={"data__date__fr_FR__isnull":true,"data__date__en_GB__isnull":false}
```

Concerning the field that cannot be `null`, but just empty, you can use the `exact` field lookup method:

```
/api/document/?filters={"data__end_date__exact": ""}
```

**Comment by gffuma** [comment]
My specific problem is:

I have to filter the dates so i use this:

`https://minett-stories.lu/api/document/?parser=yaml&overlaps=1900-01-01,2022-12-31`
The count is `75`

But the count of `https://minett-stories.lu/api/document/` is `147` i guess that document without dates are filtered out.

How i can include them?

**Comment by danieleguido** [comment]
You're correct, the `overlaps` param needs both edges to work (it uses django `__lte` and `__gte` behind the scenes)
I'm going to check what's happening ...

**Comment by danieleguido** [comment]
it seems that JSON `null` elements are not equal to `None`: https://docs.djangoproject.com/en/4.0/topics/db/queries/#storing-and-querying-for-none

Hence, to get all documents having `null` we would use this syntax in django ORM (it works):
`Document.objects.filter(Q(data__end_date=Value('null'))|Q(data__start_date=Value('null'))).count()` = 60
the opposite:
`Document.objects.exclude(Q(data__end_date=Value('null'))|Q(data__start_date=Value('null'))).count()` = 87

however, this is not available to the Miller api
